### PR TITLE
Actually correctly fix problems caused by copying `pkg/regexp.Regexp` values

### PR DESCRIPTION
**Pull request by mtrmac**
A follow-up on https://github.com/containers/storage/pull/1621/files#r1206860402 ; see individual commit messages for details.

### CODE REVIEW

File: pkg/regexp/regexp.go [hunk 1]
```diff
@@ -10,26 +10,26 @@ import (
 // used as global variables. Using this structure helps speed the startup time
 // of apps that want to use global regex variables. This library initializes them on
 // first use as opposed to the start of the executable.
-type Regexp struct {
-	once   *sync.Once
+type Regexp = *regexpStruct
+
+type regexpStruct struct {
+	_      noCopy
```
Comment (mtrmac) [inline pkg/regexp/regexp.go:16]: I have verified, at least going by `unsafe.Sizeof`, that this has no run-time memory cost.

**Comment by rhatdan** [comment]
LGTM
@nalind @vrothberg @giuseppe @saschagrunert @umohnani8 @flouthoc PTAL

**Review by flouthoc — APPROVED** [review]
LGTM

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by nalind** [comment]
LGTM

**Pull request merged**